pkg/scheduler/factory/factory.go

```diff
@@ -1064,6 +1067,70 @@ func (c *configFactory) invalidateCachedPredicatesOnNodeUpdate(newNode *v1.Node,
 	}
 }
 
+func nodeSchedulingPropertiesChanged(newNode *v1.Node, oldNode *v1.Node) bool {
+	if nodeAllocatableChanged(newNode, oldNode) {
```

Comment:
Updated the order to: `Spec.Unschedulable` -> `Status.Allocatable` -> `Labels` -> `Taints` -> `Conditions`. In general, `Status.Allocatable` is changed more frequent than `Labels` and `Taints`, so I make it as the second. `Conditions` is put at the end as it performs more operations to strip each conditions.

Reply:
The order sounds good. Condition changes are somewhat frequent though as kubelet updates last transition time of a condition somewhat frequently. So, it might be better to check that before some other ones, but without actual data it is hard to find the right order. We can go with the current order and study the impact of change of ordering later.